### PR TITLE
Docs/luc/overview diagram

### PR DESCRIPTION
**Pull request by lucpeterson**
Adds some overview documentation, including a diagram figure

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -7,8 +7,66 @@ Welcome to Merlin's documentation!
...
+`maestro <https://github.com/LLNL/maestrowf>`_ to
+provide an interface for describing workflows, as well as for defining
+workflow task dependencies. It translates those dependencies into concrete
+tasks via `celery <https://docs.celeryproject.org/>`_, 
+which can be configured for a variety of backend
+technologies (`rabbitmq <https://www.rabbitmq.com>`_ and
+`redis <https://redis.io>`_ are currently supported). Although not
+a hard dependency, we encourage the use of
+`flux <http://flux-framework.org>`_ for interfacing with
+HPC batch systems, since it can scale to very large number of jobs.
```
Comment (ben-bay) [inline docs/source/index.rst]: "to a very large number" or "to very large numbers" ..

File: README.md [hunk 2]
```diff
@@ -18,6 +18,82 @@ Welcome to the Merlin README, a condensed guide. For more in-depth Merlin
...
+
+The integrated system looks a little something like this:
+
+<img src="docs/images/merlin_arch.png" width="400" alt="a typical Merlin workflow">
+
+In this example, here's how it all works:
+
+1. The scientist describes her HPC workflow as a maestro DAG (directed acyclic graph)
+"spec" file `workflow.yaml`
+2. She then sends it to the persistent server `merlin run workflow.yaml` .
```
Comment (ben-bay) [inline README.md]: "...to the persistent server with"

File: docs/source/index.rst (hunk 1 above)
Comment (lucpeterson) [reply docs/source/index.rst]: addressed

File: README.md (hunk 2 above)
Comment (lucpeterson) [reply README.md]: fixed

**Pull request merged**